frontend/src/pages/CompetitionDetails.js

```diff
@@ -303,6 +344,26 @@ const CompetitionDetails = () => {
 
   const renderPRs = () => {
     const userCreatedPRs = userPRs.filter(pr => pr.user.login === username);
+
+    if (loading) {
+      return (
+          <div style={{ display: 'flex', flexDirection: 'column', justifyContent: 'center', alignItems: 'center', height: '400px' }}>
+            {/* Your existing loader */}
+            <ClipLoader color="#36D7B7" size={50} loading={loading} />
+
+            {/* Typewriter effect */}
+            <div style={{ marginTop: '20px', color: '#36D7B7', fontFamily: 'Courier New', fontSize: '20px' }}>
+              <Typewriter
+                  words={['Loading your data...', 'Fetching PRs...', 'Please wait...']}
+                  loop={true}
+                  typeSpeed={70}
+                  deleteSpeed={50}
+                  delaySpeed={1500}
+              />
+            </div>
+          </div>
+      );
+    }
```

Comment:
you could use the same here that way you are reducing the amount of code written while keep it simple and DRY